utils/useAuth.js

```diff
@@ -1,44 +1,59 @@
-import { useEffect, useState } from 'react'
+import React, { createContext, useContext, useEffect, useState } from 'react'
 import { useRouter } from 'next/router'
 import { getMe, deleteMe } from './auth'
+import PropTypes from 'prop-types'
 
-const useAuth = () => {
+const AuthContext = createContext()
+
+const AuthProvider = (props) => {
   const router = useRouter()
-  const [token, setToken] = useState()
-  const [user, setUser] = useState()
+  const [user, setUser] = useState(null)
   const [loaded, setLoaded] = useState(false)
 
   useEffect(() => {
     const initAuth = async () => {
       try {
-        const [tokenFromServer, userFromServer] = await getMe()
-
-        if (!tokenFromServer || !userFromServer) {
-          console.log('user or token undefined')
-          await deleteMe()
-          router.push('/login')
-        } else {
-          setToken(tokenFromServer)
-          setUser(userFromServer)
-          setLoaded(true)
-        }
+        const initUser = await getMe(props.token)
+        setUser(initUser)
+        setLoaded(true)
       } catch (error) {
-        console.log('useAuth', error)
+        console.log(error)
+        setLoaded(true)
       }
     }
 
-    if (document.cookie) {
-      initAuth()
+    if (!props.token) {
+      setUser(null)
+      setLoaded(true)
     } else {
-      router.push('/login')
+      initAuth()
+    }
+  }, [props.token])
+
+  const signIn = async () => {
+    router.push('/api/spotify-auth')
+  }
+
+  const signOut = async () => {
+    try {
+      await deleteMe()
+      setUser(false)
+    } catch (error) {
+      console.log('signout issue', error)
+      setUser(null)
     }
-  }, [router])
+  }
+
+  const { token } = props
+  const data = { loaded, token, user }
 
-  return [
-    loaded,
-    token,
-    user
-  ]
+  return <AuthContext.Provider value={{ data, signIn, signOut }} {...props} />
```

Comment:
I'm not sure we need all of the data lumped together, might be easier to just have it at the top level of the object? Doesn't make much of a difference at this point really?